2_process.R

```diff
@@ -62,6 +69,26 @@ p2_targets_list <- list(
     p2_site_list_nontidal_csv,
     create_site_list_nontidal(p2_wqp_SC_filtered,p1_nwis_sites,p1_daily_data,p1_inst_data,
                               hucs=drb_huc8s,crs_out="NAD83",p2_sites_w_segs,"2_process/out/DRB_SC_sitelist_nontidal.csv"),
-    format = "file")
-
+    format = "file"),
+  
+  # # target for NADP 
+  tar_target(p2_NADP_Data,
+             lapply(list.files(path = p1_NADP_data_unzipped, full.names = T), function(x) read.csv(x, sep = ',') %>%
+                    # select only cols starting with cat and COMID co
+                    select(COMID | starts_with('CAT')) %>%
+                    # take only COMIDS in drb
+                    filter(COMID %in% p2_drb_comids_all_tribs$comid) %>%
+                    # add year col to ID each dataset - using regex to extract the year between NADP_ and _CONUS
+                    mutate(Year = str_extract_all(x, "(?<=unzipped/NADP_).+(?=_CONUS.txt)")) %>%
```

Comment:
The class of the `Year` column in the attribute table is a list. Could you change to a numeric vector?